### ISSUE TITLE
Upgrading catalog app without changing anything changes chart version to latest

### ISSUE THREAD

**Issue by bentastic27** [rancher/rancher#27891]
**What kind of request is this (question/bug/enhancement/feature request):**

**Steps to reproduce:**

1. Spin up Rancher and a custom cluster.
2. Install an app that has older chart versions available and use the older version, I used wordpress.
3. Seemingly optional: wait for it to be ready.
4. "Upgrade" the chart but don't change any values (notice that the version in the dropdown stays the same).

**Result:**

Notice that the chart version was automatically changed to the latest available.

**Other details that may be helpful:**

It doesn't seem to matter if it's from our library or from a custom one, I've tried bitnami's as well for example:

https://charts.bitnami.com/bitnami

**Environment information**

Rancher 2.3.6, 2.3.8 and 2.4.5 are all versions that I've tried.

Kubernetes 1.17.4 or .6 on a custom cluster with DO nodes.

gzrancher/rancher#10954

**Comment by jiaqiluo** [comment]
This bug is reproduced in Rancher: `v2.4.6-rc3`

Steps:
- run  Rancher: `v2.4.6-rc3` single install
- add a DO cluster
- go to the default project, deploy the app `WordPress` from the built-in library with an older version 

<img width="1412" alt="Screen Shot 2020-07-10 at 9 50 53 AM" src="https://user-images.githubusercontent.com/6218999/87179603-df445580-c293-11ea-834c-62ebe4677803.png">


- after the app is active, go to the edit page of the app
- click on the upgrade button with NO change for any value


Results:
the app is upgrade to its latest version 

Basically, when you "Upgrade" the app but don't change any values, a POST request is sent and it contains the wrong `exernalID` which seems to be always the one for the latest version of the chart.

<img width="735" alt="Screen Shot 2020-07-10 at 9 51 35 AM" src="https://user-images.githubusercontent.com/6218999/87179346-7066fc80-c293-11ea-991b-a340a35a0f01.png">

**Comment by Lebvanih** [comment]
Seems to be the same issue that I reported here a few days ago: #27828
For me the issue isn't consistent, and can be workaround by reslecting the version (this seems to trigger some calls in the background)

**Comment by SinisterMinister** [comment]
We've been getting bitten by this bug for months with the most recent bite taking down our Artifactory cluster. If the unexpected upgrade makes a change to data that can't be undone (e.g. database migration), there is no way to roll back. Is there any eta on when to expect a fix?

**Comment by hatharom** [comment]
I have the same issue, but the other way around.
For me the smallest version number will be deployed all the time.

**Comment by jiaqiluo** [comment]
The bug fix is validated on 
- v2.4-4278-head
- v2.5-head 71da15
- master-head 5aa391

the following is validated for both the **regular app** and the **multiple-cluster app** 
- When going to the upgrade page of the app, the latest version will be selected by default, and we can upgrade the app to the target version if clicking the upgrade button. 
![image](https://user-images.githubusercontent.com/6218999/99109769-8b7dac00-25a6-11eb-82af-848036d9198b.png)


- If we can change the template version back to the current version, clicking the upgrade button will Not trigger any upgrade.
